ledger/accountdb.go

```diff
@@ -682,7 +1802,7 @@ func removeEmptyAccountData(tx *sql.Tx, queryAddresses bool) (num int64, address
 // the full AccountData because we need to store a large number of these
 // in memory (say, 1M), and storing that many AccountData could easily
 // cause us to run out of memory.
-func accountDataToOnline(address basics.Address, ad *basics.AccountData, proto config.ConsensusParams) *ledgercore.OnlineAccount {
+func accountDataToOnline(address basics.Address, ad ledgercore.AccountData, proto config.ConsensusParams) *ledgercore.OnlineAccount {
```

Comment:
why did we discard the pointer?

Reply:
In this case the pointer is meaningless, since this method is going to get inlined, and therefore the stack representation isn't important anymore.

Reply:
Can we force the compiler to inline it? I'm sure that there is a way to do it
if we can't force it, I think we should pass this parameter as a pointer so in case it is not inlined we will not copy the structure 

BTW
I've tried to use `go tool nm ~/go/bin/algod | grep accountDataToOnline` and it looks like the compiler did create a function for this (I think it means that it is not inlined)

Reply:
maybe because it returns a pointer? I've seen an article about a vscode extension for golang that parses the extended compiler output and shows gc some memory allocation and inlining decision made by the compiler, but can't find the article :(

Reply:
that's is why I think it is a bit risky to count on the compiler to optimize it.
looks like it didn't do that 
we can actually use `go tool objdump -s github.com/algorand/go-algorand/ledger.accountDataToOnline ~/go/bin/algod
`
and see the actual implementation of the function in assmbly

Reply:
ok, merged changes.